SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -70,9 +72,44 @@ internal EnumCacheManager(string enumTypeName, int startingIndex, IEnumerable<in
                 }
 #endif
             }
+            else if (enumTypeName.CompareTo("CraftTreeType") == 0)
+            {
+#if DEBUG
+                Logger.Log("DEBUG: Now loading known CraftTreeTypes...");
+#endif
+                int addedToBanList = 0;
+                var enumValues = Enum.GetValues(typeof(CraftTree.Type));
+                foreach (var enumValue in enumValues)
+                {
+                    if (enumValue != null)
+                    {
+                        int realEnumValue = (int)enumValue;
+#if DEBUG
+                        Logger.Log("DEBUG: Found known CraftTreeType: ID=[" + realEnumValue + "]");
+#endif
+                        if (realEnumValue > startingIndex)
+                        {
+                            if (!this.bannedIndices.Contains(realEnumValue))
+                            {
+                                this.bannedIndices.Add(realEnumValue);
+                                ++addedToBanList;
+                            }
+#if DEBUG
+                            else
+                                Logger.Log("DEBUG: CraftTreeType ID=[" + realEnumValue + "] already present in ban list.");
+                        }
+                        else
+                            Logger.Log("DEBUG: CraftTreeType ID=[" + realEnumValue + "] is one of the game CraftTreeType.");
+#else
+                        }
+#endif
+                    }
+                }
+                Logger.Log("DEBUG: Finished known CraftTreeType exclusion. " + addedToBanList + " ID were added in ban list.");
+            }
         }
 
-        #region Caching
+#region Caching
```

Comment:
I like the spaces in the regions hashes, would be nice if you can put them back.